modules/auxiliary/gather/zoomeye_search.rb

```diff
@@ -152,27 +140,126 @@ def run
     if @zoomeye_token.blank?
       print_error("Unable to login api.zoomeye.org")
       return
+    else
+      print_status("Logged in to zoomeye")
     end
 
-    # create ZoomEye request parameters
-    dork = datastore['ZOOMEYE_DORK']
-    resource = datastore['RESOURCE']
-    page = 1
-    maxpage = datastore['MAXPAGE']
+    results = []
+    results[0] = dork_search(resource, dork, 1, facets)
 
+    if results[0]['total'].nil? || results[0]['total'] == 0
+      msg = "No results."
+      if results[0]['error'].to_s.length > 0
+        msg << " Error: #{results[0]['error']}"
+      end
+      print_error(msg)
+      return
+    end
+
+    # Determine page count based on total results
+    if results[0]['total'] % 20 == 0
+      tpages = results[0]['total'] / 20
+    else
+      tpages = results[0]['total'] / 20 + 1
+    end
+    maxpage = tpages if datastore['MAXPAGE'] > tpages
+
+    print_status("Total: #{results[0]['total']} on #{tpages} " +
+      "pages. Showing: #{maxpage} page(s)")
+
+    # If search results greater than 20, loop & get all results
+    if results[0]['total'] > 20
+      print_status('Collecting data, please wait...')
+      page = 1
+      while page < maxpage
+        page_result = dork_search(resource, dork, page+1, facets)
+        if page_result['matches'].nil?
+          next
+        end
+        results[page] = page_result
+        page += 1
+      end
+    end
+    tbl1 = Rex::Text::Table.new(
+      'Header'  => 'Search Results',
+      'Indent'  => 1,
+      'Columns' => ['IP:Port', 'City', 'Country', 'Hostname', 'OS', 'Service:Version', 'Info']
+    )
+    tbl2 = Rex::Text::Table.new(
+      'Header'  => 'Search Results',
+      'Indent'  => 1,
+      'Columns' => ['IP', 'Site', 'City', 'Country', 'DB:Version', 'WebApp:Version']
+    )
     # scroll max pages from ZoomEye
-    while page <= maxpage
-      print_status("ZoomEye #{resource} Search: #{dork} - page: #{page}")
-      results = dork_search(dork, resource, page) if dork
-      break unless match_records?(results)
-
-      matches = results['matches']
-      if resource.include?('web')
-        parse_web_records(matches)
+    results.each do |page|
+      if facets
+        fac = page['facets']
+        fac.each do |fa|
+          print_line("#{fa[0]}")
+          fa[1].each do |f|
```

Comment:
Can this be unpacked into something more descriptive rather than `fa[#]`? Something like `fac.each do |name, items|`. I'm not sure if the first element is a name or not, but whatever it is would help the reader understand what data is expected at that point.